client/src/components/badge/types.d.ts

```diff
@@ -4,4 +4,5 @@ export type BadgeProps = {
   className?: string;
   removable?: boolean;
   onClick?: (data: BadgeProps['data']) => void;
+  background?: boolean;
```

Comment:
This is fine, but for the naming, I would prefer to be more explicit on the functionality instead of the appearance. To me in this case, the `background` property doesn't have any functionality and it's better to allow add `className` from outside and the dev can decide to extend the styles as needed. Another valid option it's to define themes or variants.